From 5dcc4593b6764f0afb3b570d834e9e729ab8f63a Mon Sep 17 00:00:00 2001
From: Gautham Kumar Senthilkumaran <quic_gauthamk@quicinc.com>
Date: Mon, 26 Aug 2024 12:14:58 +0530
Subject: [PATCH] ath12k: wifi: Disabled PPE DS in Marina

Disabled the PPE DS configuration in Marina bring-up branch as currently
this feature is not supported in Marina. Crash was seen during the initial
bootup in Marina with PPE DS enabled and below is the trace seen in AP.

pc : edma_cfg_rx_ring_en_mapped_queues+0x2c/0xd0 [qca_nss_dp]
lr : edma_ppeds_inst_start+0x298/0x2f0 [qca_nss_dp]
sp : ffffff8007cfbb30
x29: ffffff8007cfbb30 x28: ffffff8007c72fa0 x27: ffffffc07928e000
x26: ffffff8007cfbc14 x25: ffffff8007c73b60 x24: ffffff8007c72fa0
x23: ffffff8007cfbc14 x22: 0000000000000001 x21: 0000000000000000
x20: ffffffc079267e98 x19: ffffffc0792a7e4c x18: ffffffc080c40e90
x17: 3231687461207665 x16: 6463206874697720 x15: 00000000000000c0
x14: 0000000000000003 x13: 0000000000000000 x12: 00000000a81969c0
x11: ffffffc087acc000 x10: 0000000000007fe0 x9 : ffffff801e057fe0
x8 : ffffffc087ad3fe0 x7 : 00000000000007ff x6 : 00000000000007ff
x5 : ffffff8007c721f0 x4 : 0000000000000000 x3 : ffffffc081b00000
x2 : 0000000000000001 x1 : 0000000000000001 x0 : ffffffc079267e98
Call trace:
 edma_cfg_rx_ring_en_mapped_queues+0x2c/0xd0 [qca_nss_dp]
 edma_ppeds_inst_start+0x298/0x2f0 [qca_nss_dp]
 ppe_ds_wlan_instance_start+0x194/0x22c [qca_nss_ppe_ds]
 ath12k_dp_ppeds_start+0x98/0x118 [ath12k]
 ath12k_core_qmi_firmware_ready+0xab0/0xdd0 [ath12k]
 ath12k_qmi_qdss_trace_mem_info_send_sync+0x7c8/0x998 [ath12k]


Signed-off-by: Gautham Kumar Senthilkumaran <quic_gauthamk@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/core.c b/drivers/net/wireless/ath/ath12k/core.c
index 90614a9..95518fd 100644
--- a/drivers/net/wireless/ath/ath12k/core.c
+++ b/drivers/net/wireless/ath/ath12k/core.c
@@ -57,7 +57,7 @@ static unsigned int ath12k_en_fwlog = true;
 module_param_named(en_fwlog, ath12k_en_fwlog, uint, 0644);
 MODULE_PARM_DESC(en_fwlog, "fwlog: 0-disable, 1-enable");
 
-unsigned int ath12k_ppe_ds_enabled = true;
+unsigned int ath12k_ppe_ds_enabled = false;
 module_param_named(ppe_ds_enable, ath12k_ppe_ds_enabled, uint, 0644);
 MODULE_PARM_DESC(ppe_ds_enable, "ppe_ds_enable: 0-disable, 1-enable");
 
-- 
2.34.1

